core/src/config.rs

```diff
@@ -19,7 +19,7 @@ use crate::{
 };
 
 /// Configuration parameters container.
-#[derive(Clone, Default, Deserialize, Serialize, Debug, Configurable, PartialEq, Eq)]
```

Comment:
why?

Reply:
When the configurations were a POD datatype, they could be hashed, compared for equality, among other things. 

Now the configuration (at least for the logger) contains both interior mutable state, as well as a `SideEffect` which cannot (and should not) be possible to compare for equality. So it's no longer true that in-general the configurations are `PartialEq` if you need this functionality for something, you can manually derive it on the configurations that have this property. 

Moreover, this is not a property that we made use of. There are no instances of code that broke because of lack of `PartialEq`.